examples/glium_gl_area/main.rs

```diff
@@ -0,0 +1,280 @@
+                        vColor = color;
+                    }
+                ",
+
+                fragment: "
+                    #version 140
+                    in vec3 vColor;
+                    out vec4 f_color;
+                    void main() {
+                        f_color = vec4(vColor, 1.0);
+                    }
+                "
+            },
+
+            110 => {
+                vertex: "
+                    #version 110
+                    uniform mat4 matrix;
+                    attribute vec2 position;
+                    attribute vec3 color;
+                    varying vec3 vColor;
+                    void main() {
+                        gl_Position = vec4(position, 0.0, 1.0) * matrix;
+                        vColor = color;
+                    }
+                ",
+
+                fragment: "
+                    #version 110
+                    varying vec3 vColor;
+                    void main() {
+                        gl_FragColor = vec4(vColor, 1.0);
+                    }
+                ",
+            },
+
+            100 => {
+                vertex: "
+                    #version 100
+                    uniform lowp mat4 matrix;
+                    attribute lowp vec2 position;
+                    attribute lowp vec3 color;
+                    varying lowp vec3 vColor;
+                    void main() {
+                        gl_Position = vec4(position, 0.0, 1.0) * matrix;
+                        vColor = color;
+                    }
+                ",
+
+                fragment: "
+                    #version 100
+                    varying lowp vec3 vColor;
+                    void main() {
+                        gl_FragColor = vec4(vColor, 1.0);
+                    }
+                ",
+            },
+        )
+        .unwrap();
+
+        Some(Renderer {
+            context,
+            vertex_buffer,
+            index_buffer,
+            program,
+        })
+    }
+
+    fn draw(&self) {
+        let mut frame = self.context.draw();
+
+        let uniforms = uniform! {
+            matrix: [
+                [1., 0., 0., 0.],
+                [0., 1., 0., 0.],
+                [0., 0., 1., 0.],
+                [0., 0., 0., 1f32]
+            ]
+        };
+
+        frame.clear_color(0., 0., 0., 0.);
+        frame
+            .draw(
+                &self.vertex_buffer,
+                &self.index_buffer,
+                &self.program,
+                &uniforms,
+                &Default::default(),
+            )
+            .unwrap();
+        frame.finish().unwrap();
+    }
+}
+
+fn main() {
+    // Load GL pointers from epoxy (GL context management library used by GTK)
+    {
+        #[cfg(unix)]
+        let library = libloading::os::unix::Library::this();
+        #[cfg(windows)]
+        let library =
+            libloading::os::windows::Library::open_already_loaded("libepoxy-0.dll").unwrap();
```

Comment:
I think on Linux you ideally also want to use libepoxy to make sure the same GL libraries are used as GTK is using.

Reply:
Yeah, that's what happens with this code. The `epoxy` crate loads epoxy-specific symbols which are available in the global address space.

Reply:
But `libloading::os::unix::Library::this()` would load whatever is currently loaded, which might not be what GTK is using? Or am I missing something?

Reply:
GTK is always using epoxy, and the `epoxy` crate loads specifically `epoxy_` symbols as far as I can tell: https://github.com/mjkoo/epoxy-rs/blob/ebb7328dbd7f6dd394ffb80e6d6b4e5696b8d800/build/epoxy_gen.rs#L226

Reply:
Yes but you currently only use epoxy on Windows, not elsewhere

Reply:
On Linux the symbols can be `dlsym`'d without opening `epoxy` specifically (which means one less string to hardcode). Not sure about macOS.

Reply:
My point is that on Linux this also depends a lot on the environment. You global `glFoo` symbol at this point might not be the one that GTK is using. This is regularly a problem with GL on platforms like the RPi or with proprietary NVIDIA drivers.

Is there anything wrong with just using epoxy unconditionally everywhere?